### PR TITLE
Provide default for uncertainty type keyword to ensure pre-3.1 files can be read

### PR DESCRIPTION
**Pull request by mwcraig**
This addresses https://github.com/astropy/ccdproc/issues/664, which arises because a check was added in 3.1 for uncertainty type. Though part of the check allowed for the possibility that it was reading an older file, if the uncertainty type was missing from the FITS header the read would fail. This adds a regression test and fixes the issue.

### CODE REVIEW

**Comment by mwcraig** [comment]
Not sure whether to tag this 3.2 or 3.1.3 (it can and should be backported to 3.1.3 if there is another 3.1.x release).

**Comment by bsipocz** [comment]
@mwcraig - I'm not sure whether we'll have one more 3.1.x release, but nevertheless for milestoning we should assumer there would be (and if not, it would only mean that I need to do a bit more of changelog cleaning before release than usual).

File: astropy/nddata/tests/test_ccddata.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     InverseVariance)
 from astropy import units as u
 from astropy import log
+from astropy.io import fits
```
Comment (pllim) [inline astropy/nddata/tests/test_ccddata.py]: Is this import necessary?

Comment (mwcraig) [reply astropy/nddata/tests/test_ccddata.py]: Good catch, it is not. Was going to use it to generate the file I needed for testing, then realized I could do it another way.

File: astropy/nddata/tests/test_ccddata.py [hunk 2]
```diff
@@ -879,6 +879,34 @@ def test_write_read_multiextensionfits_custom_ext_names(ccd_data, tmpdir):
...
+def test_read_old_style_multiextensionfits(tmpdir):
+    # Regression test for https://github.com/astropy/ccdproc/issues/664
+    #
+    # Prior to astropy 3.1 there was no uncertainty type saved
+    # in the multiextension fits files generated by CCDData
+    # because the uncertainty had to be StandardDevUncertainty.
+    #
+    # Current version should be able to read those in.
+    #
+    size = 100
```
Comment (pllim) [inline astropy/nddata/tests/test_ccddata.py]: Is 100x100 array really necessary? If the content does not matter, why not do `np.arange(4).reshape((2, 2))` so you can save on memory and avoid a call to random generator? (Sorry, should have attached it to my previous review.)

Comment (mwcraig) [reply astropy/nddata/tests/test_ccddata.py]: Thanks for the suggestion, fixed.

**Pull request merged**